tests/integration/records/single_db/test_records_unload_datetime.py

```diff
@@ -0,0 +1,203 @@
+            self.records.move(source, target)
+            from records_mover.records import mover
+            mover.logger.info('Done with move')
+            directory_loc = self.session.directory_url(directory_url)
+            records_dir = RecordsDirectory(records_loc=directory_loc)
+            with tempfile.NamedTemporaryFile() as t:
+                output_url = pathlib.Path(t.name).as_uri()
+                output_loc = self.session.file_url(output_url)
+                records_dir.save_to_url(output_loc)
+                return output_loc.string_contents()
+
+    def test_unload_date(self) -> None:
+        self.createDateTable()
+        for dateformat in DATE_CASES:
+            addl_hints: PartialRecordsHints = {}
+            if self.engine.name == 'redshift':
+                if dateformat != 'YYYY-MM-DD':
+                    # this is the only format supported by Redshift on
+                    # export, so we're going to need to be sure to use
+                    # hints that work in Pandas
+                    addl_hints = {
+                        'datetimeformat': f'{dateformat} HH24:MI:SS',
+                        'datetimeformattz': f'{dateformat} HH:MI:SS',
+                    }
+            records_format = RecordsFormat(variant=VARIANT_FOR_DB[self.engine.name],
+                                           hints={
+                                               'dateformat': dateformat,
+                                               'compression': None,
+                                               'header-row': False,
+                                               **addl_hints,  # type: ignore
+                                           })
+            csv_text = self.unload(column_name='date',
+                                   records_format=records_format)
+            self.assertEqual(csv_text, create_sample(dateformat) + "\n",
+                             f"from dateformat {dateformat} and addl_hints {addl_hints}")
+
+    def test_unload_datetime(self) -> None:
+        self.createDateTimeTable()
+        matching_dateformat = {
+            'YYYY-MM-DD HH:MI:SS': 'YYYY-MM-DD',
+            'YYYY-MM-DD HH12:MI AM': 'YYYY-MM-DD',
+            'MM/DD/YY HH24:MI': 'MM/DD/YY',
+        }
+        for datetimeformat in DATETIME_CASES:
+            addl_hints: PartialRecordsHints = {}
+            if self.engine.name == 'redshift':
+                if datetimeformat != 'YYYY-MM-DD HH24:MI:SS':
+                    # this is the only format supported by Redshift on
+                    # export, so we're going to need to be sure to use
+                    # hints that work in Pandas - i.e.,
+                    addl_hints = {
+                        'dateformat': matching_dateformat[datetimeformat],
+                        'datetimeformattz': datetimeformat,
+                    }
+                    if 'AM' in datetimeformat:
+                        # TODO: Add a GitHub issue for this
```

Comment:
Uncompleted punchlist item detected--consider resolving or moving this to your issue tracker